### PR TITLE
LibWeb: Add support for grid-auto-flow

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**

### CODE REVIEW

**Review by awesomekling — APPROVED** [review]

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
`||` strikes again! :smiling_face_with_tear:

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 1]
```diff
@@ -5308,6 +5309,60 @@ RefPtr<StyleValue> Parser::parse_grid_track_size_list(Vector<ComponentValue> con
...
+            return GridAutoFlowStyleValue::Dense::Yes;
+        }
+        return {};
+    };
+
+    if (auto axis = parse_axis(); axis.has_value()) {
+        if (auto dense = parse_dense(); dense.has_value()) {
+            return GridAutoFlowStyleValue::create(*axis, *dense);
+        } else {
+            return GridAutoFlowStyleValue::create(*axis, GridAutoFlowStyleValue::Dense::No);
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:5353]: If the "dense" component fails to parse, e.g `grid-auto-flow: row green;`, shouldn't we reject it here?

Comment (kalenikaliaksandr) [reply Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp]: indeed, we should! I fixed that.

File: Userland/Libraries/LibWeb/CSS/StyleProperties.cpp [hunk 2]
```diff
@@ -901,6 +902,15 @@ CSS::GridTrackSizeList StyleProperties::grid_template_rows() const
     return value->as_grid_track_size_list().grid_track_size_list();
 }
 
+CSS::GridAutoFlow StyleProperties::grid_auto_flow() const
+{
+    auto value = property(CSS::PropertyID::GridAutoFlow);
+    if (value->type() != CSS::StyleValue::Type::GridAutoFlow)
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/StyleProperties.cpp:908]: Nit-pick: `if (!value->is_grid_auto_flow())`

File: Userland/Libraries/LibWeb/CSS/StyleValue.h [hunk 3]
```diff
@@ -235,6 +237,7 @@ class StyleValue : public RefCounted<StyleValue> {
     GridTrackPlacementStyleValue const& as_grid_track_placement() const;
     GridTrackSizeListShorthandStyleValue const& as_grid_track_size_list_shorthand() const;
     GridTrackSizeListStyleValue const& as_grid_track_size_list() const;
+    GridAutoFlowStyleValue const& as_grid_auto_flow() const;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/StyleValue.h:240]: We also want a non-const version of this, somewhere below.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Parsing looks good now. :+1:

**Pull request merged**